daemon/rpc_set_allowlist.go

```diff
@@ -11,12 +11,12 @@ import (
 	"github.com/NordSecurity/nordvpn-linux/internal"
 )
 
-// containsPrivateNetwork returns true if subnets contains a private network not conainted in oldWhitelist
+// containsPrivateNetwork returns true if subnets contains a private network
 func containsPrivateNetwork(subnets []string) bool {
 	for _, subnet := range subnets {
 		if net, err := netip.ParsePrefix(subnet); err != nil {
 			log.Println("Failed to parse subnet: ", err)
-		} else if net.Addr().IsPrivate() {
+		} else if net.Addr().IsPrivate() || net.Addr().IsLinkLocalUnicast() {
```

Comment:
There were several places where we incorrectly wouldn't be testing for link local addresses. We consider such addresses as private when forwarding traffic or enabling lan discovery.